homeassistant/components/qbus/entity.py

```diff
@@ -0,0 +1,101 @@
+"""Base class for Qbus entities."""
+
+import re
+
+from qbusmqttapi.discovery import QbusMqttOutput
+from qbusmqttapi.factory import QbusMqttMessageFactory, QbusMqttTopicFactory
+from qbusmqttapi.state import QbusMqttState
+
+from homeassistant.components.mqtt import client as mqtt
+from homeassistant.components.mqtt.models import ReceiveMessage
+from homeassistant.helpers.device_registry import DeviceInfo, format_mac
+from homeassistant.helpers.entity import Entity
+
+from .const import DOMAIN
+from .qbus import QbusEntry
+
+_REFID_REGEX = r"^\d+\/(\d+(?:\/\d+)?)$"
```

Comment:
pre-compile regex for re-use
```suggestion
_REFID_REGEX = re.compile(r"^\d+\/(\d+(?:\/\d+)?)$")
```